common/ellipsis.js

```diff
@@ -217,64 +213,83 @@
-                    var updateHeight = function(index, element) {
-                        var height = element.clientHeight;
-                        if (height > maxHeight) {
-                            scope.overflow[index] = true;
-                            scope.hideContent = true;
-                            containsOverflow = true;
-                            scope.maxHeightStyle = maxHeightStyle;
-                        } else {
-                            scope.overflow[index] = false;
+                            // +10 to account for padding on TD
+                            scope.overflow[i] = (currentTD.children[0].clientHeight + 10) > maxHeight
                         }
                     }
 
-                    // Resizerow is called whenever there is a change in rowValues model
-                    // It iterates over all the td elements and extracts image and iframes from it
-                    // After which it binds onload event to adjust height
-                    // It also calls updateHeight for the same td, for any oveflown textual content
-                    var resizeRow = function() {
-                        if (containsOverflow == false) {
-
-                            //Iterate over table data in the row
-                            for (var i = 0; i < element[0].children.length; i++) {
-
-                                // Get all images and iframes inside the td
-                                var imagesAndIframes = UiUtils.getImageAndIframes(element[0].children[i]);
-
-                                // Bind onload event and updateheight for particular td index
-                                imagesAndIframes.forEach(function(el) {
-                                    var index = i;
-                                    el.onload = function() {
-                                        updateHeight(index, el);
-                                    };
-                                });
+                    var initializeOverflowLogic = function () {
+                        setOverflows();
+
+                        scope.hideContent = true;
+                        scope.maxHeightStyle = maxHeightStyle;
+
+                        scope.$digest();
+                    }
 
-                                updateHeight(i, element[0].children[i].children[0]);
+                    new ResizeSensor(element[0], function (dimensions) {
+                        // if TR.offsetHeight > the calculated maxRecordsetRowHeight
+                        // +10 to account for padding on TD element
+                        if (dimensions.height > (maxHeight + 10)) {
+                            // iterate over each cell (TD), check it's height, and set overflow if necessary
+                            if (!userClicked) {
+                                initializeOverflowLogic();
                             }
+                        } else if (dimensions.height < (maxHeight + 10)) {
```

Comment:
Instead of adding just 10 here, you should define constant variables somewhere so it's easier to find and change them. You should name (or comment) them in such a way that it makes it easier to map it to the css rule that is defining these paddings.